velero-plugins/pod/restore.go

```diff
@@ -50,8 +50,19 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	if err != nil {
 		return nil, err
 	}
-	// Check if pod has owner Refs
-	if len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" {
+
+	// get backup associated with the restore
+	backupName := input.Restore.Spec.BackupName
+	backup, err := common.GetBackup(backupName, input.Restore.Namespace)
+	if err != nil {
+		return nil, err
```

Comment:
If we can't get a valid backup, should we fail here (possibly providing a restore failure unrelated to the root cause), or should we just not consider the defaultVolumesToRestic scenario if we don't get a valid backup?

Reply:
I like Scott's approach of not considering this scenario if it fails. Seems less prone to flakes

Reply:
Right. Basically the end result of this change should be that if we can determine that default volumes to restic is enabled, then we keep the pod if there are volumes. if the flag is false, or it's nil, or anything goes wrong in trying to determine any of the above, then we just do what we did before this PR -- skip the pod if it's owned and doesn't have a restic annotation.

Reply:
So the easiest way to do this would be to declare `defaultVolumesToResticFlag` as a `*bool` var (leaving it nil), then remove the error return and instead:
```
if err == nil {
	// check for default restic flag
	defaultVolumesToResticFlag := backup.Spec.DefaultVolumesToRestic

}

Reply:
that way this will be nil if getting the backup fails, which means the logic below considers it the same as false

Reply:
updated the PR